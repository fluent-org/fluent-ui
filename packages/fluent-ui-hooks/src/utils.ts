export const global = typeof window !== 'undefined' && window
