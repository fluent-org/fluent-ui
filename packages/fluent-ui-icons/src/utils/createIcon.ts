import * as React from 'react'
import styled from 'styled-components'

export type Children = [string, any]
export type JSX = ['svg', any, Children]

function createIcon(jsx: JSX, componentName: string): React.ForwardRefExoticComponent<{}> {
  const [type, props, children] = jsx
  function Icon(_props: any): any {
    return React.createElement(type, { ...props, ..._props }, React.createElement(...children))
  }
  const SvgIconStyled = styled(Icon)`
    width: 0.83em;
    height: 0.83em;
    display: inline-block;
    user-select: none;
    font-size: inherit;
    color: inherit;
    fill: currentColor;
  `
  SvgIconStyled.displayName = componentName
  return SvgIconStyled
}

export default createIcon
