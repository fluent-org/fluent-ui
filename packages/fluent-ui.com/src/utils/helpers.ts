export function getCookie(name: string): string {
  const regex = new RegExp(`(?:(?:^|.*;*)${name}*=*([^;]*).*$)|^.*$`)
  return document.cookie.replace(regex, '$1')
}
