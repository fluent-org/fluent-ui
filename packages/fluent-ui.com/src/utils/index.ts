export * from './scroll'
export * from './theme'
