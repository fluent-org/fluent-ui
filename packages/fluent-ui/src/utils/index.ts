export { isObject, deepMerge } from './deepMerge'
export { default as omit } from './omit'
export * from './mixins'
