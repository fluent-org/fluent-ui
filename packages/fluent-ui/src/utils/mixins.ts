export const findAllByType = (component: any, type: string): any[] => {
  let matches: any[] = []

  if (component.type === type) {
    matches.push(component)
  }

  if (component.children) {
    component.children.forEach(
      (child: any): void => {
        matches = matches.concat(findAllByType(child, type))
      }
    )
  }

  return matches
}
