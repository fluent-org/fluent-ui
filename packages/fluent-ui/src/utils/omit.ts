interface Obj {
  [key: string]: any
}

function omit(obj: Obj, fields: string[]): Omit<Obj, string> {
  const shallowCopy: Obj = {
    ...obj
  }
  for (let i = 0; i < fields.length; i++) {
    const key: string = fields[i]
    delete shallowCopy[key]
  }
  return shallowCopy
}

export default omit
