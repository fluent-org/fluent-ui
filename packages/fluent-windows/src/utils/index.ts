export * from './mixins'
export * from './convertingColor'
