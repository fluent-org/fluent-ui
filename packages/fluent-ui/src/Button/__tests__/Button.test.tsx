import * as React from 'react'
import * as renderer from 'react-test-renderer'
import 'jest-styled-components'

import { ThemeProvider, Button } from '../..'
import { findAllByType } from '../../utils'

describe('Button', (): void => {
  const theme = {}
  test('basic', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button>basic</Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('variant', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button variant="primary">primary</Button>
        <Button variant="accent">accent</Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('disabled', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button disabled>disabled</Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('size', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button size="small">small</Button>
        <Button size="medium">medium</Button>
        <Button size="large">large</Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('onClick', (): void => {
    const onClick = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button onClick={onClick}>onClick</Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()

    const button = findAllByType(tree, 'button')
    button[0].props.onClick()
    expect(onClick).toHaveBeenCalled()
  })

  test('as', (): void => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Button as="a" href="#">
          as a
        </Button>
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
