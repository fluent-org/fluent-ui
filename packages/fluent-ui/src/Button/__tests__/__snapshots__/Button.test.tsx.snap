// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render children 1`] = `
".FButton-root {
  border: 2px solid;
  cursor: pointer;
  display: inline-block;
  outline: none;
  min-width: 56px;
  text-align: center;
  transition: background-color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,border-color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,box-shadow 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,padding 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms;
  line-height: 1.5;
  border-radius: 2px;
  vertical-align: middle;
  text-decoration: none;
}
.FButton-root:disabled {
  color: #8c8c8c;
  cursor: not-allowed;
  pointer-events: none;
  background-color: #e9e9e9;
}
.FButton-variantStandard {
  color: #000000;
  border-color: #dedede;
  background-color: #ffffff;
}
.FButton-variantStandard:hover, .FButton-variantStandard:focus {
  border-color: #b5b5b5;
}
.FButton-variantStandard:active {
  border-color: #595959;
  background-color: #dedede;
}
.FButton-variantPrimary {
  color: #ffffff;
  box-shadow: 0px 6.4px 14.4px 0px rgba(0,0,0,0.132),0px 1.2px 3.6px 0px rgba(0,0,0,0.108);
  border-color: #0078d4;
  background-color: #0078d4;
}
.FButton-variantPrimary:hover, .FButton-variantPrimary:focus {
  border-color: #005cad;
  background-color: #005cad;
}
.FButton-variantPrimary:active {
  box-shadow: 0px 12.8px 28.8px 0px rgba(0,0,0,0.132),0px 2.4px 7.2px 0px rgba(0,0,0,0.108);
  border-color: #002d61;
  background-color: #002d61;
}
.FButton-sizeSmall {
  padding: 0 8px;
}
.FButton-sizeMedium {
  padding: 3px 12px;
}
.FButton-sizeLarge {
  padding: 8px 42px;
}
.FButton-block {
  width: 100%;
}
.FButton-ghost {
  color: inherit;
  border-color: rgba(0, 0, 0, 0.1);
  background-color: transparent;
}
.FButton-ghost:hover, .FButton-ghost:focus {
  border-color: rgba(0, 0, 0, 0.2);
  background-color: transparent;
}
.FButton-ghost:active {
  border-color: rgba(0, 0, 0, 0.4);
  background-color: transparent;
}"
`;
